### PR TITLE
Allow domain sockets to configure SO_SNDBUF and SO_RCVBUF

### PR DESCRIPTION
**Pull request by ellisjoe**
Motivation:

Running tests with a `KQueueDomainSocketChannel` showed worse performance than an `NioSocketChannel`. It turns out that the default send buffer size for Nio sockets is 64k while for KQueue sockets it's 8k. I verified that manually setting the socket's send buffer size improved perf to expected levels.

Modification:

Plumb the `SO_SNDBUF` and `SO_RCVBUF` options into the `DomainSocketChannelConfig`.

Result:

Can now configure send and receive buffer sizes for domain sockets.

### CODE REVIEW

File: transport-native-unix-common/src/main/java/io/netty/channel/unix/DomainSocketChannelConfig.java [hunk 1]
```diff
@@ -77,4 +79,24 @@
      * Return the {@link DomainSocketReadMode} for the channel.
      */
     DomainSocketReadMode getReadMode();
```
Comment (normanmaurer) [inline transport-native-unix-common/src/main/java/io/netty/channel/unix/DomainSocketChannelConfig.java]: We can’t make these changes in a point release :( That said we can keep the ones on the actual implementation. Please revert this part

File: transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollDomainSocketChannelConfig.java [hunk 2]
```diff
@@ -52,6 +53,12 @@
         if (option == ALLOW_HALF_CLOSURE) {
             return (T) Boolean.valueOf(isAllowHalfClosure());
         }
+        if (option == SO_SNDBUF) {
```
Comment (normanmaurer) [inline transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollDomainSocketChannelConfig.java]: The same also needs to be done for setOption

File: transport-native-kqueue/src/main/java/io/netty/channel/kqueue/KQueueDomainSocketChannelConfig.java [hunk 3]
```diff
@@ -52,6 +53,12 @@
         if (option == ALLOW_HALF_CLOSURE) {
             return (T) Boolean.valueOf(isAllowHalfClosure());
         }
+        if (option == SO_SNDBUF) {
+            return (T) Integer.valueOf(getSendBufferSize());
```
Comment (normanmaurer) [inline transport-native-kqueue/src/main/java/io/netty/channel/kqueue/KQueueDomainSocketChannelConfig.java]: The same also needs to be done for setOption

File: transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollDomainSocketChannelConfig.java [hunk 4]
```diff
@@ -24,9 +24,10 @@
 import io.netty.channel.unix.DomainSocketChannelConfig;
 import io.netty.channel.unix.DomainSocketReadMode;
 
+import java.io.IOException;
 import java.util.Map;
 
-import static io.netty.channel.ChannelOption.ALLOW_HALF_CLOSURE;
+import static io.netty.channel.ChannelOption.*;
```
Comment (normanmaurer) [inline transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollDomainSocketChannelConfig.java]: please revert this... we usually not do wildcard imports for such things

File: transport-native-unix-common/src/main/java/io/netty/channel/unix/DomainSocketChannelConfig.java (hunk 1 above)
Comment (ellisjoe) [reply transport-native-unix-common/src/main/java/io/netty/channel/unix/DomainSocketChannelConfig.java]: removed. is there somewhere you'd like to track adding this in a future release?

**Comment by ellisjoe** [comment]
thanks for the review @normanmaurer. I signed the icla and addressed your comments. Let me know if there's anything else you'd like me to follow up with.

**Comment by normanmaurer** [comment]
I will just do it directly in master as a followup

File: transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollDomainSocketChannelConfig.java [hunk 5]
```diff
@@ -173,4 +186,42 @@ public EpollDomainSocketChannelConfig setAllowHalfClosure(boolean allowHalfClosu
         this.allowHalfClosure = allowHalfClosure;
         return this;
     }
+
+    @Override
```
Comment (normanmaurer) [inline transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollDomainSocketChannelConfig.java]: You need to remove the `@Override` annotations now

File: transport-native-kqueue/src/main/java/io/netty/channel/kqueue/KQueueDomainSocketChannelConfig.java [hunk 6]
```diff
@@ -159,6 +173,44 @@ public DomainSocketReadMode getReadMode() {
         return mode;
     }
 
+    @Override
```
Comment (normanmaurer) [inline transport-native-kqueue/src/main/java/io/netty/channel/kqueue/KQueueDomainSocketChannelConfig.java]: You need to remove the `@Override` annotations now

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by normanmaurer** [comment]
@ellisjoe thanks a lot!

**Comment by normanmaurer** [comment]
@ellisjoe followup is here https://github.com/netty/netty/pull/9602

**Pull request merged**